sql/sql_show.cc

```diff
@@ -2137,6 +2142,184 @@ class List_process_list : public Do_THD_Impl {
+           inspect_thd->get_protocol()->connection_alive()) &&
+         !inspect_thd->system_thread) ||
+        (m_user && (inspect_thd->system_thread || !inspect_sctx_user.str ||
+                    strcmp(inspect_sctx_user.str, m_user)))) {
+      mysql_mutex_unlock(&inspect_thd->LOCK_thd_protocol);
+      return;
+    }
+    mysql_mutex_unlock(&inspect_thd->LOCK_thd_protocol);
+
+    thread_info *thd_info = new (*THR_MALLOC) thread_info;
+
+    /* ID */
+    thd_info->thread_id = inspect_thd->thread_id();
+
+    /* USER */
+    if (inspect_sctx_user.str)
+      thd_info->user = m_client_thd->mem_strdup(inspect_sctx_user.str);
+    else if (inspect_thd->system_thread)
+      thd_info->user = "system user";
+    else
+      thd_info->user = "unauthenticated user";
+
+    /* HOST */
+    if (inspect_thd->peer_port &&
+        (inspect_sctx_host.length || inspect_sctx->ip().length) &&
+        m_client_thd->security_context()->host_or_ip().str[0]) {
+      if ((thd_info->host =
+               (char *)m_client_thd->alloc(LIST_PROCESS_HOST_LEN + 1)))
+        snprintf((char *)thd_info->host, LIST_PROCESS_HOST_LEN, "%s:%u",
+                 inspect_sctx_host_or_ip.str, inspect_thd->peer_port);
+    } else
+      thd_info->host = m_client_thd->mem_strdup(
+          inspect_sctx_host_or_ip.str[0]
+              ? inspect_sctx_host_or_ip.str
+              : inspect_sctx_host.length ? inspect_sctx_host.str : "");
+
+    DBUG_EXECUTE_IF("processlist_acquiring_dump_threads_LOCK_thd_data", {
+      if (inspect_thd->get_command() == COM_BINLOG_DUMP ||
+          inspect_thd->get_command() == COM_BINLOG_DUMP_GTID)
+        DEBUG_SYNC(m_client_thd,
+                   "processlist_after_LOCK_thd_list_before_LOCK_thd_data");
+    });
+    /* DB */
+    mysql_mutex_lock(&inspect_thd->LOCK_thd_data);
+    const char *db = inspect_thd->db().str;
+    if (db) thd_info->db = m_client_thd->mem_strdup(db);
+    mysql_mutex_unlock(&inspect_thd->LOCK_thd_data);
+
+    auto time_stats = get_thd_time_stats(inspect_thd);
+    /* Statement_seconds */
+    double2my_decimal(E_DEC_FATAL_ERROR, std::get<0>(time_stats),
+                      &thd_info->stmt_secs);
+    thd_info->stmt_secs.frac = 6;
+    /* Transaction_seconds */
+    double2my_decimal(E_DEC_FATAL_ERROR, std::get<1>(time_stats),
+                      &thd_info->trx_secs);
+    thd_info->trx_secs.frac = 6;
+    /* Command_seconds */
+    double2my_decimal(E_DEC_FATAL_ERROR, std::get<2>(time_stats),
+                      &thd_info->cmd_secs);
+    thd_info->cmd_secs.frac = 6;
+
+    /* rw_trans */
+    thd_info->rw_trans = inspect_thd->rw_trans;
+    /* sql_log_bin */
+    thd_info->sql_log_bin = inspect_thd->variables.sql_log_bin;
+
+    m_thread_infos->push_back(thd_info);
+  }
+};
+
+void mysqld_list_transactions(THD *thd, const char *user, bool verbose) {
+  Item *field;
+  List<Item> field_list;
+  Thread_info_array thread_infos(thd->mem_root);
+  size_t max_query_length =
```

Comment:
Calculated here, and passed to List_transaction_list's constructor but never used inside the class.